### PR TITLE
Bluetooth: Host: Fix ISO buffers leak

### LINKED ISSUE #77006: Bluetooth: Host: ISO buffer not unreffed
** Describe the bug **

When sending data using ISO and the data is fragmented, if the connection is cut before all the fragments are sent, the data buffer will not get destroyed and will be leaked.

** To reproduce **

Create an ISO broadcaster that will sent fragmented SDU. Terminate the ISO BIS after the first fragment is sent. Repeat that until you run out of buffers.

The following BabbleSim test reproduce the issue: https://github.com/theob-pro/zephyr/commit/e2d8225f7cba3b2fa3748dd7cd33a83b32db0326

** Expected behavior **

The data buffers should not leak when the ISO connection is lost.

### LINKED ISSUE #77007: Bluetooth: Host: ISO not clearing TX context on disconnect
** Describe the bug **

When disconnected while sending data, if ISO doesn't get the num of completed packets it will not call `process_unack_tx` and thus will leak TX context.

Fix that by setting the connection state in ISO disconnection which will trigger a call to `process_unack_tx`.

** To reproduce **

Create an ISO channel that will send data, disconnect before receiving the HCI number of completed packets event.

The following BabbleSim test reproduce the issue: https://github.com/theob-pro/zephyr/commit/e2d8225f7cba3b2fa3748dd7cd33a83b32db0326

### PR DESCRIPTION
**Pull request by theob-pro**
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/77006
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/77007

Add a test to reproduce those issues.

### CODE REVIEW

**Review by alwa-nordic — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Review by jori-nordic — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

File: tests/bsim/bluetooth/host/iso/frag_2/src/broadcaster.c [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+	 * - no buffer is leaked and repeating the operation success
+	 */
+	int err;
+	uint8_t NUM_BUF = 4;
+
+	LOG_INF("Starting ISO HCI fragmentation test 2");
+
+	init();
+
+	for (int l = 0; l < NUM_BUF; l++) {
```
Comment (Thalley) [inline tests/bsim/bluetooth/host/iso/frag_2/src/broadcaster.c:229]:
```suggestion
	for (int i = 0; i < NUM_BUF; i++) {
```

Why are you using `l` like some kind of madman :D

**Review by Thalley — COMMENTED** [review]
Does this need to get backported?

File: tests/bsim/bluetooth/host/iso/frag_2/src/broadcaster.c (hunk 1 above)
Comment (jori-nordic) [reply tests/bsim/bluetooth/host/iso/frag_2/src/broadcaster.c:229]: l == loop
i == iteration
dolores: `Doesn't seem like anything to me`

Comment (jhedberg) [reply tests/bsim/bluetooth/host/iso/frag_2/src/broadcaster.c:229]: i == iterator
l == looper? :)

Come on - let's just stick to the what's common and not trigger extra cycles in people's brains when they read the code.

**Review by jori-nordic — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Pull request merged**